projects/common-api/src/client/java/dan200/computercraft/api/client/turtle/TurtleUpgradeModeller.java

```diff
@@ -34,6 +37,20 @@ public interface TurtleUpgradeModeller<T extends ITurtleUpgrade> {
      */
     TransformedModel getModel(T upgrade, @Nullable ITurtleAccess turtle, TurtleSide side);
 
+    /**
+     * Obtain the model to be used when rendering a turtle peripheral.
+     * <p>
+     * Used when turtle access object doesn't exist. For compatibility reasons, by default this method call getModel
+     * that depend on turtle.
+     *
+     * @param data  Upgrade data instance for current turtle side
+     * @param side    Which side of the turtle (left or right) the upgrade resides on.
+     * @return The model that you wish to be used to render your upgrade.
+     */
+    default TransformedModel getModel(@Nonnull UpgradeData<T> data, TurtleSide side) {
```

Comment:
I think we should probably unpack the `UpgradeData` here and make this `T upgrade, CompoundTag tag,` to match the other method.

Reply:
Done